operators/api/v1alpha2/template_types.go

```diff
@@ -136,6 +139,19 @@ type EnvironmentResources struct {
 	Disk resource.Quantity `json:"disk,omitempty"`
 }
 
+// ContainerStartupOpts specifies custom startup options for the created container,
+// including the possibility to download and extract a zip file to a given destination
+// and specifying the arguments that will be passed to the application container.
+type ContainerStartupOpts struct {
+	// URL from which GET the zip file to be extracted into ContentPath
+	SourceZipPURL string `json:"sourceZipURL,omitempty"`
```

Comment:
Would personally avoid to force using zip files if unnecessary. Apparently, the `unar` utility is able to uncompress a wide variety of format, including zip, rar, 7z, tar, ...